migration/clean-up-scripts/add_vault_prefix.py

```diff
@@ -0,0 +1,57 @@
+import subprocess
+import argparse
+import json
+import sys
+
+parser = argparse.ArgumentParser("Vault Prefixer", "Provide a path to a list of vault UUIDs.", "Prefixes all vaults passed to the script from the specified file with '!'.")
+parser.add_argument("--file", action="store", dest="filepath", help="Required. Specify the path to a file containing a linebreak-delimited list of vault UUIDs you'd like to process.", required=True)
+parser.add_argument("-r", action="store_true", dest="deletemode", help="Use this option to delete vaults in the provided list, rather than append `!` to the vault name. USE WITH CAUTION.")
+args = parser.parse_args()
+
+# Prefix each vault in the input list with an exclamation mark to make them easy to locate and assess duplicate status. {}
+def prefixVault():
+    print("Prefixing all listed vaults with '!' for easy sorting and identification.\n\n")
+    with open(args.filepath, "r", encoding="utf-8") as vaultList:
+        for vault in vaultList:
+            vaultID = vault.rstrip()
+            try:
+                vaultDetails = json.loads(subprocess.run(f"op vault get {vaultID} --format=json", shell=True, check=True, capture_output=True).stdout)
+                vaultName = vaultDetails['name']
+            except (Exception) as err:
+                print(f"Encountered an error getting the details for vault with ID {vaultID}: ", err)
```

Comment:
I might have been poorly Pythoning, but this script errored out seemingly weirdly at this point:

```
Encountered an error getting the details for vault with ID vaultName,vaultUUD,itemCount,vaultCreated,vaultUpdated,userCount:  Command 'op vault get vaultName,vaultUUD,itemCount,vaultCreated,vaultUpdated,userCount --format=json' returned non-zero exit status 1.
Encountered an error getting the details for vault with ID bippity,ddsweci7li5jubu4izcnkfaj4y,0,2023-9-1 22:30:6,2023-9-1 22:30:6,0:  Command 'op vault get bippity,ddsweci7li5jubu4izcnkfaj4y,0,2023-9-1 22:30:6,2023-9-1 22:30:6,0 --format=json' returned non-zero exit status 1.
...
```

Reply:
Woah that is weird. The exception message definitely suggests something very strange happened. 

did this occur when you ran it with a service account? Did this also happen when you used a 1Password account in the Owners group?

Reply:
Oh, hang on, I think I understand now. I suspect your list of vault IDs contained additional information? 

The script expects a file like this:

```
usrzroiqicsryaa3o56nddixu4
pyxedwd5d47phwh72l6kyp5h7a
wupizr5o5z4vrehdjsaaicr2cu
```

which I realize is somewhat narrowly defined, but the use-case would be that once you've isolated the candidates for deletion from the csv created by the helper script that you'd copy the uuids into a file that contains only those UUUIds. 

However, would it be better if the script expected a file formatted exactly like the csv? The assumption being the user deleted all rows except the candidates for deletion? 


My only concern about that is it could lead someone to run the script against an _unmodified_ version of the csv file that would delete _all_ vaults as a result. Requiring them to copy and paste the desired UUIDs will force an extra bit of work to prevent a potential mistake. 

I'm on the fence and could easily be swayed one way or another.

Reply:
I think I had a different understanding of what the scripts were intended to do, but now that I understand a little better, my two cents is that someone could filter out rows of the CSV output from the helper script based on whatever criteria they liked and then save it with the headers and columns intact for use in _this_ script.

Definitely some documentation will help whichever way you go. :-)

Reply:
I'd like to provide an off-the-shelf experience that caters to the majority of users the majority of the time. 

Do you think that most script users would want to filter the output at the _script level_, rather than once the csv has been generated? 

Or do you think that most users would most often want to manipulate the CSV using their preferred spreadsheet software?